hack/aws-acceptance-test-cleanup/main.go

```diff
@@ -759,6 +764,68 @@ func destroyBackoff(ctx context.Context, resourceKind string, resourceID string,
 	}, backoff.WithContext(expoBackoff, ctx))
 }
 
+func cleanupIAMRoles(ctx context.Context, iamClient *iam.IAM) error {
+	var rolesToDelete []*iam.Role
+	rolePrefix := "consul-k8s-"
+	err := iamClient.ListRolesPagesWithContext(ctx, &iam.ListRolesInput{},
+		func(page *iam.ListRolesOutput, lastPage bool) bool {
+			for _, role := range page.Roles {
+				roleName := aws.StringValue(role.RoleName)
+				if strings.HasPrefix(roleName, rolePrefix) {
+					rolesToDelete = append(rolesToDelete, role)
+				}
+			}
+			return !lastPage
+		})
+	if err != nil {
+		return fmt.Errorf("failed to list roles: %v", err)
+	}
+
+	if len(rolesToDelete) == 0 {
+		fmt.Println("Found no iamRoles to clean up")
+		return nil
+	}
+
+	// Delete filtered roles
+	for _, role := range rolesToDelete {
+		roleName := aws.StringValue(role.RoleName)
+		err := detachRolePolicies(iamClient, role.RoleName)
+		if err != nil {
+			fmt.Printf("Failed to detach policies for role %s: %v", roleName, err)
+			continue
+		}
+
+		_, err = iamClient.DeleteRole(&iam.DeleteRoleInput{
+			RoleName: role.RoleName,
+		})
+		if err != nil {
+			fmt.Printf("Failed to delete role %s: %v", roleName, err)
+		} else {
+			fmt.Printf("Deleted role: %s", roleName)
+		}
+	}
+
+	return nil
+}
+
+func detachRolePolicies(iamClient *iam.IAM, roleName *string) error {
+	err := iamClient.ListAttachedRolePoliciesPages(&iam.ListAttachedRolePoliciesInput{
+		RoleName: roleName,
+	}, func(page *iam.ListAttachedRolePoliciesOutput, lastPage bool) bool {
+		for _, policy := range page.AttachedPolicies {
+			_, err := iamClient.DetachRolePolicy(&iam.DetachRolePolicyInput{
+				RoleName:  roleName,
+				PolicyArn: policy.PolicyArn,
+			})
+			if err != nil {
+				fmt.Printf("Failed to detach policy %s from role %s: %v", *policy.PolicyArn, *roleName, err)
```

Comment:
do we need to worry about the nil pointer deref here?

Reply:
As we are not de-ref the `err` and just check the nullity of the var, I think it's OK. Please correct me if I were wrong.

And the PR looks all good from my side! Thank you for your great work Anita!

Reply:
the issue is on `*roleName` and `*policy.PolicyArn` which could potentially be a nil pointer

Reply:
They are coming from `page.AttachedPolicies` and `page.Roles` which are offered by the `aws-sdk-go` package, I checked the source, both are list of pointers. So maybe we could *trust* aws? 😂

Reply:
it's probably better to not  trust that we're getting good intput and code defensively here, aws provides functions for dealing with pointers and we should probably use them here to avoid the potential nil derefs https://docs.aws.amazon.com/sdk-for-go/api/aws/#StringValue

Reply:
Yeah, I saw that and used that function to deref those values. Thanks for pointing it out